### PR TITLE
Automation Test refactoring and support for UIBehavior picker

### PR DESCRIPTION
**Pull request by bgavrilMS**
1. Add more state to objects instead of passing state in each method
2. Remove signing on MSAL test to be able to take dependency on Xamarin.UITest

### CODE REVIEW

File: core/tests/Test.Microsoft.Identity.Core.UIAutomation/CoreMobileTestHelper.cs [hunk 1]
```diff
@@ -33,37 +33,44 @@
...
 {
     public class CoreMobileTestHelper
-    {   
-        public void PerformSignInFlow(ITestController controller, IUser user)
+    {
+        ITestController _controller;
+
+        public CoreMobileTestHelper(ITestController testController)
+        {
+            this._controller = testController;
```
Comment (jennyf19) [inline core/tests/Test.Microsoft.Identity.Core.UIAutomation/CoreMobileTestHelper.cs]: do we still need "this"?

File: msal/tests/Test.MSAL.NET.UIAutomation/MSALMobileTestHelper.cs [hunk 2]
```diff
@@ -39,64 +35,74 @@ namespace Test.MSAL.UIAutomation
...
         /// Runs through the standard acquire token silent flow
         /// </summary>
         /// <param name="controller">The test framework that will execute the test interaction</param>
-        public void AcquireTokenSilentTestHelper(ITestController controller, UserQueryParameters userParams)
+        public void AcquireTokenSilentTestHelper(UserQueryParameters userParams)
         {
             //acquire token for 1st resource
-            AcquireTokenInteractiveHelper(controller, userParams);
-            _coreMobileTestHelper.VerifyResult(controller);
+            AcquireTokenInteractiveHelper(userParams, "login");
```
Comment (jennyf19) [inline msal/tests/Test.MSAL.NET.UIAutomation/MSALMobileTestHelper.cs]: can this be made a const ?

File: msal/tests/Test.MSAL.NET.UIAutomation/XamarinDroidMSALTests.cs [hunk 3]
```diff
@@ -43,14 +43,14 @@ namespace Test.MSAL.UIAutomation
...
-        MSALMobileTestHelper _msalMobileTestHelper = new MSALMobileTestHelper();
+        IApp _app;
+        Platform _platform;
+        ITestController _xamarinController;
+        MSALMobileTestHelper _msalMobileTestHelper;
 
         public XamarinMSALDroidTests(Platform platform)
         {
-            this.platform = platform;
+            this._platform = platform;
```
Comment (jennyf19) [inline msal/tests/Test.MSAL.NET.UIAutomation/XamarinDroidMSALTests.cs:53]: remove "this"?

**Review by jennyf19 — APPROVED** [review]
nice changes. 🦈

File: adal/tests/Test.ADAL.NET.UIAutomation/XamarinDroidADALTests.cs [hunk 4]
```diff
@@ -44,14 +44,14 @@ namespace Test.ADAL.UIAutomation
...
-        ADALMobileTestHelper _adalMobileTestHelper = new ADALMobileTestHelper();
+        IApp _app;
+        Platform _platform;
+        ITestController _testController;
+        ADALMobileTestHelper _adalMobileTestHelper;
 
         public XamarinDroidADALTests(Platform platform)
         {
-            this.platform = platform;
+            this._platform = platform;
```
Comment (henrik-me) [inline adal/tests/Test.ADAL.NET.UIAutomation/XamarinDroidADALTests.cs:54]: >this [](start = 12, length = 4)

...

File: adal/tests/Test.ADAL.NET.UIAutomation/XamarinDroidADALTests.cs [hunk 5]
```diff
@@ -142,7 +144,9 @@ public void AcquireTokenInteractiveADFSv3NonFederatedTest()
             user.FederationProvider = FederationProvider.AdfsV3;
             user.IsFederatedUser = false;
 
-            _adalMobileTestHelper.AcquireTokenInteractiveTestHelper(xamarinController, user);
+            _adalMobileTestHelper.AcquireTokenInteractiveTestHelper(user);
         }
+
+
```
Comment (henrik-me) [inline adal/tests/Test.ADAL.NET.UIAutomation/XamarinDroidADALTests.cs:150]: ?

File: core/tests/Test.Microsoft.Identity.Core.UIAutomation/StringExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+//------------------------------------------------------------------------------
+
+using System.Text;
+
+namespace Test.Microsoft.Identity.Core.UIAutomation
+{
+    public static class StringExtensions
+    {
+     
+
```
Comment (henrik-me) [inline core/tests/Test.Microsoft.Identity.Core.UIAutomation/StringExtensions.cs:35]: ?

File: core/tests/Test.Microsoft.Identity.Core.UIAutomation/CoreUiTestConstants.cs [hunk 7]
```diff
@@ -66,9 +66,15 @@ public static class CoreUiTestConstants
...
         public const string SettingsPageID = "Settings";
         public const string ScopesEntryID = "scopesList";
+        public const string UiBehaviorID = "uiBehavior";
 
         //Test Constants
         public const int ResultCheckPolliInterval = 1000;
         public const int MaximumResultCheckRetryAttempts = 20;
+
+        // UIBehavior - these should match the public MSAL constants
+        public const string UIBehviorLogin = "login";
```
Comment (jennyf19) [inline core/tests/Test.Microsoft.Identity.Core.UIAutomation/CoreUiTestConstants.cs:76]: sp: UIBehaviorLogin

File: core/tests/Test.Microsoft.Identity.Core.UIAutomation/CoreUiTestConstants.cs [hunk 8]
```diff
@@ -66,9 +66,15 @@ public static class CoreUiTestConstants
...
+        public const string UIBehviorConsent = "consent";
```
Comment (jennyf19) [inline core/tests/Test.Microsoft.Identity.Core.UIAutomation/CoreUiTestConstants.cs:77]: >UIBehviorConsent [](start = 28, length = 16)

sp

File: core/tests/Test.Microsoft.Identity.Core.UIAutomation/CoreUiTestConstants.cs [hunk 9]
```diff
@@ -66,9 +66,15 @@ public static class CoreUiTestConstants
...
+        public const string UIBehviorSelectAccount = "select_account";
```
Comment (jennyf19) [inline core/tests/Test.Microsoft.Identity.Core.UIAutomation/CoreUiTestConstants.cs:78]: >IBehviorSelectAccount [](start = 29, length = 21)

sp

**Review by henrik-me — APPROVED** [review]
:shipit:

File: msal/tests/Test.MSAL.NET.UIAutomation/XamarinDroidMSALTests.cs [hunk 10]
```diff
@@ -59,8 +59,11 @@ public XamarinMSALDroidTests(Platform platform)
         [SetUp]
         public void InitializeBeforeTest()
         {
-            app = AppFactory.StartApp(platform, "com.Microsoft.XFormsDroid.MSAL");
-            xamarinController.Application = app;
+            _app = AppFactory.StartApp(_platform, "com.Microsoft.XFormsDroid.MSAL");
+            _xamarinController = new XamarinUITestController();
+            _msalMobileTestHelper = new MSALMobileTestHelper(_xamarinController);
+
```
Comment (trwalke) [inline msal/tests/Test.MSAL.NET.UIAutomation/XamarinDroidMSALTests.cs]: extra line not needed

Comment (trwalke) [inline msal/tests/Test.MSAL.NET.UIAutomation/XamarinDroidMSALTests.cs:65]: Extra Line not needed

File: msal/tests/Test.MSAL.NET.UIAutomation/MSALMobileTestHelper.cs [hunk 11]
```diff
@@ -39,64 +37,102 @@ namespace Test.MSAL.UIAutomation
...
+        {
+            _testController = testController;
+            _coreMobileTestHelper = new CoreMobileTestHelper(testController);
+        }
         /// <summary>
         /// Runs through the standard acquire token flow
         /// </summary>
         /// <param name="controller">The test framework that will execute the test interaction</param>
-        public void AcquireTokenInteractiveTestHelper(ITestController controller, UserQueryParameters userParams)
+        public void AcquireTokenInteractiveTestHelper(
```
Comment (trwalke) [inline msal/tests/Test.MSAL.NET.UIAutomation/MSALMobileTestHelper.cs:52]: why split this up into 3 lines?

**Review by trwalke — COMMENTED** [review]
couple of comments but otherwise looks great.

**Comment by bgavrilMS** [comment]
Too many changes to tests, will rebase and push manually.

**Pull request closed without merging**